examples/demo/src/reviews/reviewSaga.js

```diff
@@ -1,23 +1,9 @@
 import { all, put, takeEvery } from 'redux-saga/effects';
-import { push } from 'react-router-redux';
 import { showNotification } from 'react-admin';
-import {
-    REVIEW_APPROVE_SUCCESS,
-    REVIEW_APPROVE_FAILURE,
-    REVIEW_REJECT_SUCCESS,
-    REVIEW_REJECT_FAILURE,
-} from './reviewActions';
+import { REVIEW_APPROVE_FAILURE, REVIEW_REJECT_FAILURE } from './reviewActions';
 
 export default function* reviewSaga() {
     yield all([
-        takeEvery(REVIEW_APPROVE_SUCCESS, function*() {
-            yield put(
-                showNotification(
-                    'resources.reviews.notification.approved_success'
-                )
-            );
-            yield put(push('/reviews'));
-        }),
         takeEvery(REVIEW_APPROVE_FAILURE, function*({ error }) {
```

Comment:
Now I understand why we still need the saga. That means something is missing in the core sagas (`fetch` and `undo`); we should be able to handle `onFailure` side effects from the action meta, too.

Reply:
Wouldn't it be redundant with the default dataProvider error handling ?

Reply:
No, the `dataProvider` has no particular handling. It can return exceptions, that the `fetch` saga redispatches as `_FAILURE` actions. But the side effects of the fetch error (redirection, notification, refresh) could be described as data, in the initial action, just like the `onSuccess` meta.